jwst/master_background/expand_to_2d.py

```diff
@@ -0,0 +1,314 @@
+        the npixels column.
+
+    Returns
+    -------
+    background : JWST MultiSlitModel
+        A copy of `input` but with the data replaced by the background,
+        "expanded" from 1-D to 2-D.
+    """
+
+    background = input.copy()
+
+    for (k, slit) in enumerate(input.slits):
+        wl_array = get_wavelengths(slit)
+        if wl_array is None:
+            raise RuntimeError("Can't determine wavelengths for {}"
+                               .format(type(slit)))
+
+        # Wherever the wavelength is NaN, the background flux should to be set
+        # to 0.  We replace NaN elements in wl_array with -1, so that np.interp
+        # will detect that those values are out of range (note the `left`
+        # argument to np.interp) and set the output to 0.
+        wl_array[np.isnan(wl_array)] = -1.
+
+        # bkg_flux will be a 2-D array, because wl_array is 2-D.
+        bkg_flux = np.interp(wl_array, tab_wavelength, tab_background,
+                             left=0., right=0.)
+
+        background.slits[k].data[:] = bkg_flux.copy()
+
+    return background
+
+
+def bkg_for_image(input, tab_wavelength, tab_background):
+    """Create a 2-D background for an ImageModel.
+
+    Parameters
+    ----------
+    input : JWST ImageModel
+        The input science data.
+
+    tab_wavelength : 1-D ndarray
+        The wavelength column read from the 1-D background table.
+
+    tab_background : 1-D ndarray
+        The flux column read from the 1-D background table, divided by
+        the npixels column.
+
+    Returns
+    -------
+    background : JWST ImageModel
+        A copy of `input` but with the data replaced by the background,
+        "expanded" from 1-D to 2-D.
+    """
+
+    background = input.copy()
+
+    wl_array = get_wavelengths(input)
+    if wl_array is None:
+        raise RuntimeError("Can't determine wavelengths for {}"
+                           .format(type(input)))
+
+    wl_array[np.isnan(wl_array)] = -1.
+
+    # bkg_flux will be a 2-D array, because wl_array is 2-D.
+    bkg_flux = np.interp(wl_array, tab_wavelength, tab_background,
+                         left=0., right=0.)
+
+    background.data[:] = bkg_flux.copy()
+
+    return background
+
+
+def bkg_for_IFUimage(input, tab_wavelength, tab_background):
+    """Create a 2-D background for an IFUImageModel
+
+    Parameters
+    ----------
+    input : JWST IFUImageModel
+        The input science data.
+
+    tab_wavelength : 1-D ndarray
+        The wavelength column read from the 1-D background table.
+
+    tab_background : 1-D ndarray
+        The flux column read from the 1-D background table, divided by
+        the npixels column.
+
+    Returns
+    -------
+    background : JWST IFUImageModel
+        A copy of `input` but with the data replaced by the background,
+        "expanded" from 1-D to 2-D.
+    """
+
+    background = input.copy()
+    background.data[:, :] = 0.
+
+    if input.meta.instrument.name == "NIRSPEC":
```

Comment:
Might want to add .upper() to the end of instrument.name, just to make sure the value is upper-case.